vcpkg/users/binarycaching-troubleshooting.md

```diff
@@ -0,0 +1,164 @@
+* Make sure you have enough disk space on your machine
+
+### Failing to write to the binary cache
+
+A successful write to the binary cache should have the following output for each dependency:
+```
+[DEBUG] 1011: CreateProcessW("<path>\downloads\tools\nuget-6.2.1-windows\nuget.exe" push -ForceEnglishOutput -Verbosity detailed -NonInteractive "<path>\buildtrees\zlib_x86-windows.1.2.13-vcpkgbb1c96759ac96102b4b18215db138daedd3eb16c2cd3302ae7bffab2b643eb87.nupkg" -Timeout 100 -Source <NuGet Feed>)
+[DEBUG] 1011: cmd_execute_and_stream_data() returned 0 after   767027 us
+NuGet Version: 6.2.1.2
+Pushing zlib_x86-windows.1.2.13-vcpkgbb1c96759ac96102b4b18215db138daedd3eb16c2cd3302ae7bffab2b643eb87.nupkg to '<NuGet Feed>'...
+Your package was pushed.
+Stored binaries in 1 destinations in 1.8 s.
+```
+
+Your NuGet feed or cache should look something like this:
+```
+Mode                 LastWriteTime         Length Name
+----                 -------------         ------ ----
+-a---           8/16/2023  8:53 PM         529044 zlib_x86-windows.1.2.13-vcpkgbb1c96759ac96102b4b18215db138daedd3eb16c2cd3302ae7bffab2b643eb87.nupkg
+```
+
+If writing to the cache is failing, consider the following:
+
+* Make sure your binary source has `write` permissions
+* Make sure you are correctly authenticating to your provider. This may mean your authentication token has correct permissions, or, for cloud environments, you may need to run a tool to authenticate
+* Make sure `CreateProcessW` returns a successful exit code when pushing
+
+### Failing to read from the binary cache
+
+A successful reuse of a stored binary package should output:
+```
+[DEBUG] 1008: CreateProcessW("<path>\downloads\tools\nuget-6.2.1-windows\nuget.exe" install -ForceEnglishOutput -Verbosity detailed -NonInteractive "<path>\buildtrees\packages.config" -OutputDirectory "<path>\packages" -ExcludeVersion -PreRelease -PackageSaveMode nupkg -DirectDownload -NoCache -Source <NuGet Feed>)
+[DEBUG] 1008: cmd_execute_and_stream_data() returned 0 after  1168243 us
+...
+Restored 1 package(s) from NuGet in 1.2 s. Use --debug to see more details.
+```
+
+If the ABI of your package is the same between runs, make sure:
+
+* Your binary source has `read` permissions
+* You are correctly authenticating to your provider. This may mean your authentication token has correct permissions or for cloud environments, you may need to run a tool to authenticate
+
+If the ABI hash of your package differs between runs, see the next section.
+
+## Fixing your binary cache setup
+
+If you have correctly configured binary caching for both reads and writes but vcpkg is sometimes or always rebuilding your libraries instead of using existing cached binaries, read on.
+
+If your vcpkg binary cache is constantly rebuilding, it's important to figure out what changed in your [ABI hash](./binarycaching.md#abi-hash). Suppose we are building `zlib` for `x86-windows`. What information in the debug output is relevant?
```

Comment:
```suggestion
If your vcpkg binary cache is constantly rebuilding, it's important to figure out what changed in your [ABI hash](./binarycaching.md#abi-hash).
```
Avoid rhetorical questions in How-To & Troubleshooting; the objective is to solve problems as efficiently as possible and not to educate or inspire.